### PR TITLE
Delivery report for milestone one of the third grant of OpenBrush(6 milestone)

### PR DESCRIPTION
**Pull request by TtomaS7**
[Preview](https://github.com/Supercolony-net/Grant-Milestone-Delivery/blob/feature/derlivery-openbrush-follow-up-2-milestone-1/deliveries/openbrush_follow_up_2-milestone_1.md)

# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/921

### CODE REVIEW

File: deliveries/openbrush_follow_up_2-milestone_1.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/openbrush-follow-up-2.md
+* **Milestone Number:** 1 milestone of the grant and 6 milestone of the OpenBrush
```
Comment (takahser) [inline deliveries/openbrush_follow_up_2-milestone_1.md:6]:
```suggestion
* **Milestone Number:** 6
```

**Review by takahser — CHANGES_REQUESTED** [review]
@TtomaS7 is there a specific reason for submitting M6 before M5?
Also, feel free to accept my edit suggestion since this should be a numeric value (and in this case, I think it should be `6`).

**Comment by xgreenx** [comment]
In previous reports, we used "X milestone of the grant Y and Z milestone of the OpenBrush"=) But if it is okay to use just 6, we prefer that way=) I don't remember why we decided to go the previous way.

We added a small description in the delivery report, but I can provide more description.

Milestone five contains two points, which are changes in the ink!.
- "Contribute to ink! with fixing of events" - We described our thoughts about the design of new events [here](https://github.com/paritytech/ink/issues/809#issuecomment-974096940) and [here](https://github.com/paritytech/ink/pull/1243). Seems ink! team okay with our ideas, but they want to finish that task by themself.
- "Re-work the storage of contracts" - We had an active [discussion](https://github.com/paritytech/ink/issues/1134) in February and March about what new storage should look like. We finished the [initial implementation](https://github.com/paritytech/ink/issues/1134) of the storage in begin of May. After a discussion/review at the end of May, we decided to change some moments. We refactored the change. Next month ink! team were busy, so the next review took a long. Now we created a new fresh [PR](https://github.com/paritytech/ink/pull/1331) with the hope of finishing the review=)

All that already took a lot of time. We've started work on milestone six [before the creation of the grant](https://github.com/w3f/Grants-Program/pull/921#discussion_r871216213) because of milestone five. Most of the points in milestone seven are also ready(because we work on it in parallel). Soon we will start working on milestone eight(and we need to create a new grant for that, but we can't because milestone six is not delivered cause of milestone five).

Milestone five depends on ink! team and related to ink!. It is partially related to OpenBrush, but we will refactor OpenBrush when the change is merged.

We only want to unblock our work and avoid cases where all milestones in the grant are already done before the creation of that grant.

**Comment by takahser** [comment]
@xgreenx thanks for elaborating on why you deliver M6 before M5 and updating the Milestone value in the doc.
We'll have a more detailed look at the delivery soon.

**Comment by TtomaS7** [comment]
hello!
I have updated the invoice with new W3F address. 

we faced up with it on Typechain grant, so now it’s updated everywhere :D

**Comment by cruikshankss** [comment]
Hi @TtomaS7, thank you for your delivery & I wanted to update you that I began work on this evaluation today.

**Comment by cruikshankss** [comment]
Good day @TtomaS7 & @xgreenx, 

It's taken some time to deeply study your application, delivery & OpenBrush in depth, but thank you very much for the excellent article and WasmConf lecture. I'm planning to watch the Decoded lecture next & I'm almost ready to review the code. Very excited to! Thanks!

**Comment by cruikshankss** [comment]
Good day @TtomaS7 & @xgreenx,

I've completed [my evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/openbrush-follow-up_6_cruikshankss.md) of your M6 and forwarded your invoice internally. Thank you very much! 

By the way, @TtomaS7, I vaguely remember from your pitch call--I mentioned I could send you some resources but I cannot remember what these are. Now I am much more familiar with your project having completed an evaluation for you. I believe I'm much more able to help. I remember Green perhaps mentioning that there's no native inheritance in Rust but that you've done a workaround? Are you in need of any resources? By the way, excellent job with the Wasm Conference and great job managing the versioning of ink!!

**Comment by TtomaS7** [comment]
Thank you for the review!

I don't remember too :D
I will think and return to you in Element.

**Comment by cruikshankss** [comment]
@TtomaS7 You're welcome! Btw, I remembered over the weekend! And so I just sent you a ping on Element. :)

**Pull request merged**